### ISSUE TITLE
请教一下作者，复现Autoformer在exchange_rate数据集上的结果不理想

### ISSUE THREAD

**Issue by weituo2002** [thuml/Autoformer#5]
您好！
       最近有幸看到了您的Autoformer的顶会论文，下载了您在github上的源码进行实验。我用的是exchange_rate.csv数据集进行测试，测试环境的是window10+anconda+python3.6+pytorch1.9，使用了您的/scripts/Exchange_script/Autoformer.sh脚本中的第二个命令行：
       python -u run.py --is_training 1 --root_path ./dataset/exchange_rate/ --data_path exchange_rate.csv --model_id Exchange_96_192 --model Autoformer --data custom --features M --seq_len 96 --label_len 48 --pred_len 192 --e_layers 2 --d_layers 1 --factor 3 --enc_in 8 --dec_in 8 --c_out 8 --des 'Exp' --itr 1 --train_epochs 1
       得到的命令行窗口输出如下：
Use GPU: cuda:0
>>>>>>>start training : Exchange_96_192_Autoformer_custom_ftM_sl96_ll48_pl192_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_'Exp'_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 5024
val 569
test 1326
        iters: 100, epoch: 1 | loss: 0.3912162
        speed: 0.8711s/iter; left time: 50.5228s
Epoch: 1 cost time: 110.04935765266418
Epoch: 1, Steps: 157 | Train Loss: 0.3484863 Vali Loss: 0.3024623 Test Loss: 0.2717947
Validation loss decreased (inf --> 0.302462).  Saving model ...
Updating learning rate to 0.0001
>>>>>>>testing : Exchange_96_192_Autoformer_custom_ftM_sl96_ll48_pl192_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_'Exp'_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (41, 32, 192, 8) (41, 32, 192, 8)
test shape: (1312, 192, 8) (1312, 192, 8)
mse:0.27179470658302307, mae:0.3820776641368866
    
     但是在输出的可视化图形上看到预测值和实际值有不小的差距，无法复现论文中附录E.2的图Figure 12。
     我将您的源码中的exp_main.py中的test函数中可视化部分做了小小的修改，让他输出全部图形：
 #if i % 20 == 0: #全部打印
input = batch_x.detach().cpu().numpy()
gt = np.concatenate((input[0, :, -1], true[0, :, -1]), axis=0)
pd = np.concatenate((input[0, :, -1], pred[0, :, -1]), axis=0)
visual(gt, pd, os.path.join(folder_path, str(i) + '.pdf'))
   但是从图形上看真实值和预测值相差很大，例如：

![结果](https://user-images.githubusercontent.com/27657901/144353472-4e300558-c4a0-41a0-ad47-40094ff443e0.png)

   我想请教一下您，这个结果是因为我设置不对，还是理解有问题，还望您百忙之中能够给予一点指教，谢谢您了！！
   完整的可视化图形输出如下：
[result.pdf](https://github.com/thuml/Autoformer/files/7638836/result.pdf)

**Comment by wuhaixu2016** [comment]
您好！非常感谢您对Autoformer的关注。

（1）您的实验已经在数值上复现了论文结果。

（2）关于showcase，附录E.2，我们选取的case图片为输出的可视化结果的第36个序列（36.pdf）。

您可以通过降低factor的数值来增大波动，以获得论文中的结果。因为exhcange数据集预测确实比较困难，如果选择较大的facotr会引入太多的非周期噪声。希望对您有帮助。

**Comment by wuhaixu2016** [comment]
See no one has any comments on this issue, I will close it.
